libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/MapboxNavigationViewApi.kt

```diff
@@ -0,0 +1,105 @@
+import com.mapbox.navigation.ui.app.internal.destination.Destination
+import com.mapbox.navigation.ui.app.internal.destination.DestinationAction
+import com.mapbox.navigation.ui.app.internal.extension.dispatch
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationStateAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewAction
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutePreviewState
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutesAction
+import com.mapbox.navigation.ui.app.internal.tripsession.TripSessionStarterAction
+import com.mapbox.navigation.ui.app.internal.endNavigation as endNavigationAction
+
+@ExperimentalPreviewMapboxNavigationAPI
+internal class MapboxNavigationViewApi(
+    private val store: Store
+) : NavigationViewApi() {
+
+    override fun fetchRoutes(options: RouteOptions) {
+        store.dispatch(RoutePreviewAction.FetchOptions(options))
+    }
+
+    override fun fetchRoutes(points: List<Point>) {
+        store.dispatch(RoutePreviewAction.FetchPoints(points))
+    }
+
+    override fun setPreviewRoutes(routes: List<NavigationRoute>) {
+        store.dispatch(RoutePreviewAction.Ready(routes))
+    }
+
+    override fun setRoutes(routes: List<NavigationRoute>) {
+        store.dispatch(RoutesAction.SetRoutes(routes))
+    }
+
+    override fun setDestination(point: Point) {
+        store.dispatch(DestinationAction.SetDestination(Destination(point)))
+    }
+
+    override fun enableTripSession() {
+        store.dispatch(TripSessionStarterAction.EnableTripSession)
+    }
+
+    override fun enableReplaySession() {
+        store.dispatch(TripSessionStarterAction.EnableReplayTripSession)
+    }
+
+    override fun isReplayEnabled(): Boolean {
+        return store.state.value.tripSession.isReplayEnabled
+    }
+
+    override fun startFreeDrive() {
+        store.dispatch(RoutesAction.SetRoutes(emptyList()))
+        store.dispatch(RoutePreviewAction.Ready(emptyList()))
+        store.dispatch(NavigationStateAction.Update(NavigationState.FreeDrive))
+    }
+
+    @Throws(IllegalStateException::class)
+    override fun startRoutePreview() {
+        checkDestination()
+        checkPreviewRoutes()
+
+        store.dispatch(NavigationStateAction.Update(NavigationState.RoutePreview))
+    }
+
+    @Throws(IllegalStateException::class)
+    override fun startNavigation() {
+        checkDestination()
+        checkPreviewRoutes()
+
+        val routes = (store.state.value.previewRoutes as RoutePreviewState.Ready).routes
+        store.dispatch(RoutesAction.SetRoutes(routes))
+        store.dispatch(NavigationStateAction.Update(NavigationState.ActiveNavigation))
+    }
+
+    override fun startArrival() {
+        store.dispatch(NavigationStateAction.Update(NavigationState.Arrival))
+    }
+
+    override fun endNavigation() {
+        store.dispatch(endNavigationAction())
+    }
```

Comment:
Based on my discussion with @LukasPaczos he gave a good suggestion in that we don't need this function, since `endNavigation` and `startFreeDrive` are doing the same thing. So in case, someone wants to end navigation they can just invoke `startFreeDrive`.

Let's remove this function.